### PR TITLE
add lerna for monorepo package management

### PR DESCRIPTION
**Pull request by clementbalestrat**
also adds GitHub Actions CI start (a *very* simple start).

lerna has a number of advantages over npm workspaces, including:
* automatic hoisted "common" modules on the root for speed
* topographical and parallel builds/scripts
* wiring of dependant modules for convenient local development
* automatic versioning and publishing

The CI will automatically run builds, lints, and tests, and report any errors encountered.
Later on, the CI can be set up to automatically publish to NPM registry upon certain conditions (ex. when a release is cut), though this prob
is not needed/helpful with automatic versioning features provided by lerna.

This PR also cleans up some of the NPM modules arrangement and improves the linting
and building system a bit, without making too many changes.

Learn more about lerna: https://github.com/lerna/lerna

/hours 8

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,6 +12,33 @@
 | [`@synthetixio/optimism-networks`](/packages/optimism-networks)       | [![npm version](https://badge.fury.io/js/%40synthetixio%2Foptimism-networks.svg)](https://badge.fury.io/js/%40synthetixio%2Foptimism-networks)       | Network utility for Optimism Layer 2 |
 | [`@synthetixio/transaction-notifier`](/packages/transaction-notifier) | [![npm version](https://badge.fury.io/js/%40synthetixio%2Ftransaction-notifier.svg)](https://badge.fury.io/js/%40synthetixio%2Ftransaction-notifier) | Transaction utility for Layer 1 & 2  |
 
-## Repo Status
+## Developer Instructions
+
+This repo uses `lerna` to manage multiple packages in the same repo. To prepare the repository for use, run:
```
Comment (clementbalestrat) [inline README.md:17]: Thanks for adding a documentation!

**Review by clementbalestrat — COMMENTED** [review]
Great stuff @dl00 !

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  lints:
+    name: Run Lints
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v2
+      with:
+        node-version: '14'
+    - run: npm ci
```
Comment (dvd-schwrtz) [inline .github/workflows/main.yml:22]: will it work without running ci here?

**Review by dvd-schwrtz — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  lints:
+    name: Run Lints
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v2
+      with:
+        node-version: '14'
+    - run: npm ci
```
Comment (dvd-schwrtz) [inline .github/workflows/publish.yml:18]: same question as above

**Review by dvd-schwrtz — APPROVED** [review]

**Comment by KillerByte** [comment]
@clementbalestrat ready to merge when you are

**Pull request merged**